src/validation/configure.rb

```diff
@@ -41,31 +41,32 @@ def initialize(file)
       end
 
       def validate
-        @configure_results = ConfigureSchema.call(yaml: @yaml)
-        if @configure_results.success?
-          [:domain, :group, :node].each do |section|
-            @yaml[section].each do |identifier, parameters|
-              payload = {
-                section: section,
-                identifier: identifier,
-                parameters: parameters,
-              }
-              question_results = QuestionSchema.call(payload)
-              return question_results unless question_results.success?
+        @validate ||= begin
+          configure_results = ConfigureSchema.call(yaml: @yaml)
+          if configure_results.success?
+            [:domain, :group, :node].each do |section|
+              @yaml[section].each do |identifier, parameters|
+                payload = {
+                  section: section,
+                  identifier: identifier,
+                  parameters: parameters,
+                }
+                question_results = QuestionSchema.call(payload)
+                return question_results unless question_results.success?
+              end
             end
           end
+          configure_results
         end
-        @configure_results
       end
 
       def success?
-        validate if @configure_results.nil?
-        @configure_results.success?
+        validate.success?
```

Comment:
You could remove this method entirely and replace it with `delegate :success?, to: :validate` at the top of this file. You may or may not need to add `require 'active_support/core_ext/module/delegation'` to ensure `delegate` is imported.

Reply:
Fixed in: https://github.com/alces-software/metalware/pull/178/commits/f40f29e5897301f20092cb1fe8f4c7afb55978ea